packages/web-socket-browser/src/ws-browser-bus-client.ts

```diff
@@ -0,0 +1,148 @@
+    }
+
+    override async connect(
+        arg1?: ClientConnectOptions | string | number,
+        arg2?: ClientConnectOptions | string,
+        arg3?: ClientConnectOptions
+    ): Promise<void> {
+        await super.connect(arg1, arg2, arg3);
+        this._lastConnectionArgs = [ arg1, arg2, arg3 ];
+        this._isClosed = false;
+        this._isConnected = true;
+    }
+
+    override close(options?: ClientCloseOptions): Promise<void> {
+        this._isClosed = true;
+        this._isConnected = false;
+        return super.close(options);
+    }
+
+    override send(channel: string, ...args: any[]): boolean {
+        return this._isConnected && super.send(channel, ...args);
+    }
+
+    override sendTo(target: IpcBusPeer, channel: string, ...args: any[]): boolean {
+        return this._isConnected && super.sendTo(target, channel, ...args);
+    }
+
+    override request(channel: string, timeoutDelay: number, ...args: any[]): Promise<IpcBusRequestResponse> {
+        if (!this._isConnected) {
+            return Promise.reject(`Attempt to request on closed bus client ${this.peer?.id}`);
+        }
+
+        return super.request(channel, timeoutDelay, ...args);
+    }
+
+    override requestTo(
+        target: IpcBusPeer,
+        channel: string,
+        timeoutDelay: number,
+        ...args: any[]
+    ): Promise<IpcBusRequestResponse> {
+        if (!this._isConnected) {
+            return Promise.reject(`Attempt to request on closed bus client ${this.peer?.id}`);
+        }
+
+        return super.requestTo(target, channel, timeoutDelay, ...args);
+    }
+
+    override postMessage(channel: string, message: unknown, messagePorts?: BusMessagePort[]): void {
+        return this._isConnected && super.postMessage(channel, message, messagePorts);
+    }
+
+    override postMessageTo(
+        target: IpcBusPeer,
+        channel: string,
+        message: unknown,
+        messagePorts?: BusMessagePort[]
+    ): void {
+        return this._isConnected && super.postMessageTo(target, channel, message, messagePorts);
+    }
+
+    override emit(event: string, ...args: any[]): boolean {
+        return this._isConnected && super.emit(event, ...args);
+    }
+
+    override onClosed(handler: () => void): void {
+        this._closeHandler = handler;
+    }
+
+    private subscribeOnTransportClosed(): void {
+        super.onClosed(this.onTransportClosed);
+    }
+
+    private onTransportClosed(): void {
+        this._isConnected = false;
+        this._logger.info(`[WsBrowserBusClient ${this.peer?.id}] Transport closed`);
+        if (this._isClosed) {
+            this._closeHandler?.();
+        } else {
+            this._logger.info(`[WsBrowserBusClient ${this.peer?.id}] Changing state to close...`);
+            super.close();
+            this._logger.info(`[WsBrowserBusClient ${this.peer?.id}] Triggering reconnection...`);
+            this.reconnect();
+        }
+    }
+
+    private reconnect(): void {
```

Comment:
I would have considered the one time reconnection probably, because when process crashes we do not want to reconnect indefinitely...

Reply:
Limited attempts count to 20